rbc/heavydb/array.py

```diff
@@ -24,12 +29,39 @@ class HeavyDBArrayType(HeavyDBBufferType):
         bool is_null;
       }
     """
+
+    @property
+    def numba_pointer_type(self):
+        return ArrayPointer
+
     @property
     def buffer_extra_members(self):
         return ('bool is_null',)
 
 
-ArrayPointer = BufferPointer
+class ArrayPointer(BufferPointer):
+    def copy(self, context, builder, val, retptr):
+        from .buffer import memalloc
+        ptr_type = self.dtype.members[0]
+        element_size = int64_t(ptr_type.dtype.bitwidth // 8)
+
+        src = builder.extract_value(builder.load(val), 0)
+        element_count = builder.extract_value(builder.load(val), 1)
+        is_null = builder.extract_value(builder.load(val), 2)
```

Comment:
I know numba/LLVM will optimize into this result anyway, but I think the code could be a bit more readable if we just emit/declare the redundant load separately:

```suggestion
        struct_load = builder.load(val)
        src = builder.extract_value(struct_load, 0)
        element_count = builder.extract_value(struct_load, 1)
        is_null = builder.extract_value(struct_load, 2)
```